ui/e2e/explorer/route/index.spec.ts

```diff
@@ -0,0 +1,395 @@
+  ).toHaveURL(`/${namespace}/explorer/endpoint/${filename}`);
+
+  /* fill out form */
+  await page.getByLabel("path").fill("path");
+  await page.getByLabel("timeout").fill("3000");
+  await page.getByLabel("GET").click();
+  await page.getByLabel("POST").click();
+
+  /* try to save incomplete form */
+  await page.getByRole("button", { name: "Save" }).click();
+
+  await expect(
+    page.getByText("plugins : this field is invalid"),
+    "it can not save the route without a valid target plugin"
+  ).toBeVisible();
+
+  await page.getByRole("button", { name: "set target plugin" }).click();
+
+  /* add an empty instant response plugin */
+  await page.getByRole("combobox").click();
+  await page.getByLabel("Instant Response").click();
+  await page.getByRole("button", { name: "Save" }).click();
+
+  /* check editor content */
+  const editor = page.locator(".lines-content");
+  await expect(
+    editor,
+    "all entered data is represented in the editor preview"
+  ).toContainText(expectedYaml, { useInnerText: true });
+
+  /* save file */
+  await expect(
+    page.getByText("unsaved changes"),
+    "it renders a hint that there are unsaved changes"
+  ).toBeVisible();
+
+  await page.getByRole("button", { name: "Save" }).click();
+
+  await expect(
+    page.getByText("unsaved changes"),
+    "it does not render a hint that there are unsaved changes"
+  ).not.toBeVisible();
+
+  /* reload */
+  await page.reload({ waitUntil: "networkidle" });
+
+  await expect(
+    editor,
+    "after reloading, the entered data is still in the editor preview"
+  ).toContainText(expectedYaml, { useInnerText: true });
+});
+
+test("it is possible to add plugins to a route file", async ({ page }) => {
+  /* prepare data */
+  const filename = "myroute.yaml";
+  const editor = page.locator(".lines-content");
+
+  type CreateRouteYamlParam = Parameters<typeof createRouteYaml>[0];
+  const minimalRouteConfig: Omit<CreateRouteYamlParam, "plugins"> = {
+    path: "path",
+    timeout: 3000,
+    methods: ["GET", "POST"],
+  };
+
+  const basicTargetPlugin = `
+    type: "instant-response"
+    configuration:
+      status_code: 200`;
+
+  const initialRouteYaml = createRouteYaml({
+    ...minimalRouteConfig,
+    plugins: {
+      target: basicTargetPlugin,
+    },
+  });
+
+  await createWorkflow({
+    payload: initialRouteYaml,
+    urlParams: {
+      baseUrl: process.env.VITE_DEV_API_DOMAIN,
+      namespace,
+      name: filename,
+    },
+    headers,
+  });
+
+  await page.goto(`/${namespace}/explorer/endpoint/${filename}`, {
+    waitUntil: "networkidle",
+  });
+
+  /* configure inbound plugin: ACL */
+  await page.getByRole("button", { name: "add inbound plugin" }).click();
+  await page.getByRole("combobox").click();
+  await page.getByLabel("Access control list (acl)").click();
+  await page
+    .locator("fieldset")
+    .filter({ hasText: "Allow Groups (optional)" })
+    .getByPlaceholder("Enter a group")
```

Comment:
The test times out here when I try to run it locally.

Are you sure "Enter a group" is the correct placeholder to look for?

In the source code, it does not seem this translation string is used anywhere in the code: 

(pages.explorer.endpoint.editor.form.plugins.inbound.acl.groupPlaceholder)

This error occurred after merging main into the feature branch, so maybe something has changed in the mean time?

Reply:
Merging the base branch was switching that part of the form against our new `ArrayForm` and this test immediately fulfilled its purpose 😀 I just did not return the correct placeholder thre. I just pushed a fix (04e6a7643718e74005a3500ce29abea20038b0bf).